pyiron_gui/monkey_patching.py

```diff
@@ -0,0 +1,67 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+"""Adding gui functionality to pyiron classes via monkey patching on import of pyiron_gui. """
+import warnings
+
+from pyiron_gui.project.project_browser import (
+    ProjectBrowser,
+    HasGroupsBrowser,
+    DataContainerGUI,
+)
+from pyiron_base.interfaces.has_groups import HasGroups
+from pyiron_base import Project
+from pyiron_base import DataContainer
+
+__author__ = "Niklas Siemer"
+__copyright__ = (
+    "Copyright 2021, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "0.1"
+__maintainer__ = "Niklas Siemer"
+__email__ = "siemer@mpie.de"
+__status__ = "development"
+__date__ = "July 06, 2022"
+
+
+def safe_monkey_patch(cls, attr_name, value):
+    if hasattr(cls, attr_name):
+        warnings.warn(
+            f"Class {cls.__name__} already has attribute {attr_name} - Aborting monkey path of gui elements."
+        )
+    else:
+        setattr(cls, attr_name, value)
```

Comment:
This might break in a similar way under reloading module has @jan-janssen pointed out in [pyiron_base#808](https://github.com/pyiron/pyiron_base/pull/808).  I discussed a potential fix there also.